packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -3,50 +3,197 @@
+/// focus. They must be managed like other persistent state, which is typically
+/// done by a [StatefulWidget]. A stateful widget that manages a focus node must
+/// call [dispose] from its [State.dispose] method.
 ///
-/// To request focus, find the [FocusScopeNode] for the current [BuildContext]
-/// and call the [FocusScopeNode.requestFocus] method:
+/// ## Hierarchy Management
 ///
-/// ```dart
-/// FocusScope.of(context).requestFocus(focusNode);
-/// ```
+/// Because [FocusNode]s form a sparse representation of the widget tree, they
+/// must be updated whenever the widget tree is rebuilt. This is done by calling
+/// [reparentIfNeeded], usually from a [State.build] method of the widget which
+/// represents the focused region, so that the size and location of the focused
+/// area are correctly represented in its context.
 ///
-/// If your widget requests focus, be sure to call
-/// `FocusScope.of(context).reparentIfNeeded(focusNode);` in your `build`
-/// method to reparent your [FocusNode] if your widget moves from one
-/// location in the tree to another.
+/// [FocusNodes] are organized into "scopes" (see [FocusScopeNode]), which form
+/// sub-trees of nodes which can be traversed as a group. Within a scope, the
+/// most recent nodes to have focus are remembered, and if a node is focused and
+/// then removed, the original node receives focus again.
+///
+/// The focus node hierarchy can be traversed using the [parent], [children],
+/// [ancestors] and [descendants] accessors.
+///
+/// [FocusNode]s are [ChangeNotifier]s, so you can register to receive a
+/// notification when the focus changes. If you are using the [Focus] and
+/// [FocusScope] widgets to manage the nodes, it is often simpler to just
+/// establish an [InheritedWidget] dependency on them by calling [Focus.of] or
+/// [FocusScope.of] instead.
+///
+/// To see the focus tree in the debug console, you can call
+/// [debugDumpFocusTree].
 ///
 /// ## Lifetime
 ///
-/// Focus nodes are long-lived objects. For example, if a stateful widget has a
-/// focusable child widget, it should create a [FocusNode] in the
-/// [State.initState] method, and [dispose] it in the [State.dispose] method,
-/// providing the same [FocusNode] to the focusable child each time the
-/// [State.build] method is run. In particular, creating a [FocusNode] each time
-/// [State.build] is invoked will cause the focus to be lost each time the
-/// widget is built.
+/// Focus nodes are long-lived objects. For example, if a [StatefulWidget] can
+/// receive focus, and is not using a [Focus] widget, it could create a
+/// [FocusNode] in its [State.initState] method, and [dispose] it in the
+/// [State.dispose] method, providing the same [FocusNode] to the
+/// [reparentIfNeeded] call each time its [State.build] method is run. In
+/// particular, creating a [FocusNode] each time [State.build] is invoked will
+/// cause the focus to be lost each time the widget is built.
```

Comment:
this seems like a better description of what you had earlier. maybe remove the earlier text or make it more general?

we want to avoid duplicating context where possible.